src/AdminOnlyProperty/AdminOnlyPropertySendingContentNotificationHandler.cs

```diff
@@ -71,6 +71,26 @@ public void Handle(SendingContentNotification notification)
                             tab.Type = string.Empty;
                         }
                     }
+
+                    // if a Tab has only Groups and all the properties in those Groups
+                    // are now hidden we should hide the Tab too
+                    foreach (var tab in variant.Tabs.Where(t => t.Type == "Tab"))
+                    {
+                        if (tab?.Properties?.Any() == true)
+                        {
+                            continue;
+                        }
+
+                        // Groups in this Tab will have aliases that start with this Tab's alias/
+                        if (variant.Tabs.Any(t => t.Type == "Group" && t.Alias?.StartsWith(tab?.Alias + "/") == true) == true)
+                        {
+                            continue;
+                        }
```

Comment:
Initial alarm bells here, as this performs several iterations/loops.

There's the main `foreach`, which is iterating over the `variant.Tabs.Where(t => t.Type == "Tab")`, the `Where` would be performing its own iteration, (it'd be better to put this check inside the `foreach` itself.

The next loop is `variant.Tabs.Any(t => t.Type == "Group" && t.Alias?.StartsWith(tab?.Alias + "/") == true)`, is effectively a loop-within-a-loop (of the same items).

All this isn't a huge performance hit, (well depends if you have hundreds of tabs & groups).

An alternative way would be to have an additional flag in the main loop (from line 27), to pick up which groups have been hidden, taking note of the containing tab alias, then do a final loop to hide any lingering tabs.

Reply:
Thanks @leekelleher - yes I think I didn't like the extra looping, thus asking for your review. I shall give the alternative way a go 🙏